x-pack/plugins/security_solution/public/exceptions/api/list_api.ts

```diff
@@ -0,0 +1,131 @@
+import type { Rule } from '../../detection_engine/rule_management/logic';
+
+export const getListById = async ({ id, http }: FetchListById) => {
+  try {
+    const abortCtrl = new AbortController();
+    const filters = getFilters({
+      filters: { list_id: id },
+      namespaceTypes: ['single', 'agnostic'],
+      hideLists: ALL_ENDPOINT_ARTIFACT_LIST_IDS,
+    });
+    const namespaceTypes = ['single', 'agnostic'].join();
+
+    const { data } = await fetchExceptionLists({
+      filters,
+      http: http as HttpSetup,
+      signal: abortCtrl.signal,
+      namespaceTypes,
+      pagination: {},
+    });
+    abortCtrl.abort();
+
+    if (data && data.length) return data[0];
+    return null;
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+export const getListRules = async (listId: string) => {
+  try {
+    const abortCtrl = new AbortController();
+    const { data: rules } = await fetchRules({
+      signal: abortCtrl.signal,
+    });
+    abortCtrl.abort();
+    return rules.reduce((acc: Rule[], rule, index) => {
+      const listExceptions = rule.exceptions_list?.find(
+        (exceptionList) => exceptionList.list_id === listId
+      );
+      if (listExceptions) acc.push(rule);
+      return acc;
+    }, []);
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+export const updateList = async ({ list, http }: UpdateExceptionList) => {
+  try {
+    const abortCtrl = new AbortController();
+    await updateExceptionList({
+      http: http as HttpSetup,
+      list,
+      signal: abortCtrl.signal,
+    });
+    abortCtrl.abort();
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+export const unlinkListFromRules = async ({ rules, listId }: UnlinkListFromRules) => {
+  try {
+    const abortCtrl = new AbortController();
+    rules.map((rule) => {
+      const exceptionLists: ListArray | [] = (rule.exceptions_list ?? []).filter(
+        ({ list_id: id }) => id !== listId
+      );
+      return patchRule({
+        ruleProperties: {
+          rule_id: rule.rule_id,
+          exceptions_list: exceptionLists,
+        },
+        signal: abortCtrl.signal,
+      });
+    });
+    // abortCtrl.abort();
+  } catch (error) {
+    throw new Error(error);
+  }
+};
+
+export const linkListToRules = async ({
+  rules,
+  listId,
+  id,
+  listType,
+  listNamespaceType,
+}: LinkListToRules) => {
+  try {
+    const abortCtrl = new AbortController();
+    rules.map((rule) => {
+      const newExceptionList: List = {
+        list_id: listId,
+        id,
+        type: listType,
+        namespace_type: listNamespaceType,
+      };
+      const exceptionLists: ListArray | [] = [...(rule.exceptions_list ?? []), newExceptionList];
+      return patchRule({
+        ruleProperties: {
+          rule_id: rule.rule_id,
+          exceptions_list: exceptionLists,
+        },
+        signal: abortCtrl.signal,
+      });
+    });
+    // abortCtrl.abort();
```

Comment:
can this be removed?

Reply:
sure  ^_^. [353f49b](https://github.com/elastic/kibana/pull/143041/commits/353f49b20c4c7bf79fb93396fd0998d4457fcc24)